tensorflow/core/kernels/resource_variable_ops.cc

```diff
@@ -327,6 +327,18 @@ REGISTER_KERNEL_BUILDER(
 REGISTER_KERNEL_BUILDER(
     Name("VariableShape").Device(DEVICE_CPU).TypeConstraint<int64>("out_type"),
     VariableShapeOp<int64>);
+REGISTER_KERNEL_BUILDER(
+    Name("VariableShape").Device(DEVICE_DEFAULT)
+                         .TypeConstraint<int32>("out_type")
+                         .HostMemory("output")
+                         .HostMemory("input"),
```

Comment:
Nit: Can we list `input` first, just to follow the order in `REGISTER_OP`?
```suggestion
                         .HostMemory("input")
                         .HostMemory("output"),
```

Reply:
Done.